packages/react-instantsearch/src/widgets/NoResults/connect.js

```diff
@@ -9,6 +9,7 @@ export default createConnector({
     }
     return {
       noResults: search.results.nbHits === 0,
+      query: search.results.query,
```

Comment:
This is where it gets tricky (and to link to #1266), we don't want to limit ourselves to query, we might want to display more things. So it means the `<NoResults>` widget cannot have query here.

But if you need the query inside your example, we need to be able to provide an easy way to get that. And that's where the widget states and custom widget creation comes into play.

Here you would need to be able to easily create a custom widget that is able to access any InstantSearch state param. Right now this is documented but may be too hard to do.

Help :)